### PR TITLE
test: Add scripts to run kuttl tests and migrated gitlab tests to github against olm packaged operator

### PR DESCRIPTION
**Pull request by praveen4g0**
Signed-off-by: praveen4g0 <praveen.thangadancha@gmail.com>

**What type of PR is this?**

> /kind enhancement

**What does this PR do / why we need it**:
* It ensures to run kuttl tests against installed olm packaged operator, 
* ensures tests run independently doesn't have worry about adding hacks (like adding RBACS., in case ginkgo )

**Which issue(s) this PR fixes**:

This PR build the operator (operator image, bundle catalog source,  index image). t/his PR push all images to an image registry as parameter (default is the internal registry)

Fixes [#2218](https://issues.redhat.com/browse/GITOPS-2218)

**Test acceptance criteria**:
* [x] E2E Test

**How to test changes / Special notes to the reviewer**:
```
* It now adds new make target to run kuttl tests, As part of spike I have created couple of kuttl tests.
* Next step is to migrate all ginkgo tests to kuttl, will handle it in another pr

**steps to run kuttl tests**

export IMAGE="quay.io/<username>/gitops-backend-operator"
export VERSION="0.0.1"
# Now we support few ENV values to help run them locally
# 1. E2E_SKIP_OPERATOR_INSTALLATION if you set this `true`, It doesn't install operator
# 2. E2E_SKIP_BUILD_TOOL_INSTALLATION if you set this flag to `true`, It won't install operator-sdk and opm tools required to build olm bundle at runtime.
# 3. Locally you can comment out E2E_SKIP_UNINSTALL=true from `scripts/openshiftci-olm-kuttl-tests.sh` to perform uninstallation.
# Run below script to tests the changes  
# E2E_SKIP_BUNDLE_BUILD=false // Set this locally so to enable you to build and push git chnages locally, on CI we default it to true so bundling part is taken over by CI process we just export build image and version
# Run Below command
E2E_SKIP_BUNDLE_BUILD=false ./scripts/openshiftci-olm-kuttl-tests.sh
```

**Note:** As of now few tests getting failed still need to fixed!
* This script flushes out cluster dump as a tear down after execution to see argocd-operator logs at the end of our tests execution to see the status of operator

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -110,6 +110,16 @@ test-gitopsservice-nondefault:
 test: manifests generate fmt vet ## Run unit tests.
 	go test `go list ./... | grep -v test` -coverprofile cover.out
 
+.PHONY: e2e-tests-sequential
+e2e-tests-sequential:
```
Comment (wtam2018) [inline Makefile:114]: Please add a line comment after the target name so the `make help` can show the comment/help. Take a look at the `uninstall` target for example

File: Makefile [hunk 2]
```diff
@@ -130,14 +140,14 @@ install: manifests kustomize ## Install CRDs into the K8s cluster specified in ~
 	$(KUSTOMIZE) build config/crd | kubectl apply -f -
 
 uninstall: manifests kustomize ## Uninstall CRDs from the K8s cluster specified in ~/.kube/config.
-	$(KUSTOMIZE) build config/crd | kubectl delete -f -
+	$(KUSTOMIZE) build config/crd | kubectl delete --ignore-not-found=true -f -
```
Comment (wtam2018) [inline Makefile:144]: Why adding `--ignore-not-found=true` is needed?

File: Makefile [hunk 3]
```diff
@@ -196,6 +206,24 @@ OPM = $(shell which opm)
 endif
 endif
 
+.PHONY: operator-sdk
```
Comment (wtam2018) [inline Makefile:209]: Does the kuttl test need operator-sdk?

Comment (praveen4g0) [reply Makefile:209]: kuttl-tests doesn't need operator-sdk but for bundling operator we would need it

File: Makefile (hunk 2 above)
Comment (praveen4g0) [reply Makefile:144]: I saw a failure while performing make install, make deploy, make undeploy and make uninstall so added this `--ignore-not-found=true`

File: Makefile (hunk 1 above)
Comment (praveen4g0) [reply Makefile]: Done changes! Thanks

**Comment by praveen4g0** [comment]
@wtam2018 Seems master branch still uses older argocd version in [go mod file](https://github.com/redhat-developer/gitops-operator/blob/master/go.mod#L34) thats might be I see few failures with parallel tests

**Comment by iam-veeramalla** [comment]
/lgtm
/approve

Thanks @praveen4g0 for this awesome PR .. Kudos !!!

**Review by iam-veeramalla — APPROVED** [review]

**Pull request merged**